corehq/apps/cloudcare/static/cloudcare/js/formplayer/sessions/views.js

```diff
@@ -70,6 +71,76 @@ hqDefine("cloudcare/js/formplayer/sessions/views", function () {
             }
             return _.template($(id).html() || "");
         },
+        initialize: function (options) {
+            this.model = new Backbone.Model({
+                page: options.pageNumber + 1 || 1,
+                limit: parseInt($.cookie("sessions-per-page-limit")) || 10,
+            });
+            this.model.on('change', function () {
+                this.options.listSessions(this.model.get('page') - 1, this.model.get("limit"));
+                this.render.bind(this);
+            }.bind(this));
+        },
+        events: {
+            'click @ui.paginators': 'onClickPageNumber',
+            'click @ui.paginationGoButton': 'paginationGoAction',
+            'change @ui.sessionsPerPageLimit': 'onPerPageLimitChange',
+            'keypress @ui.paginationGoTextBox': 'paginationGoKeyAction',
+            'keypress @ui.paginators': 'paginateKeyAction',
+        },
+        ui: {
+            paginators: '.js-page',
+            paginationGoButton: '#pagination-go-button',
+            paginationGoTextBox: '.module-go-container',
+            paginationGoText: '#goText',
+            sessionsPerPageLimit: '.per-page-limit',
+        },
+        onClickPageNumber: function (e) {
+            e.preventDefault();
```

Comment:
what is the action that we are preventing here?

Reply:
`preventDefault` will stop any kind of browser default action that runs for a UI element when a user input event occurs i.e mouse-click or key press.

Reply:
why do we need to prevent those?

Reply:
are those links?

Reply:
Yes. The pagination actions and numbers are links.